controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -1043,10 +1043,21 @@ func (r *ReconcileHyperConverged) getSystemHealthStatus(conditions common.HcoCon
 }
 
 func isSystemHealthStatusError(conditions common.HcoConditions) bool {
+	// During upgrade progression (Progressing=true with reason HCOUpgrading),
+	// treat temporary Available=false as non-error. Only Degraded should mark error.
+	if cond, found := conditions.GetCondition(hcov1beta1.ConditionProgressing); found && cond.Status == metav1.ConditionTrue && cond.Reason == "HCOUpgrading" {
+		return conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
+	}
+
 	return !conditions.IsStatusConditionTrue(hcov1beta1.ConditionAvailable) || conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
 }
```

Comment:
The HCORequest object already contains an up-to-date boolean indication to check if we're in upgrade mode or not. It more efficient and more reliable than checking the conditions, I think.

WDYT about changing the function like this?


```suggestion
func isSystemHealthStatusError(req *common.HcoRequest) bool {
	// During upgrade progression (Progressing=true with reason HCOUpgrading),
	// treat temporary Available=false as non-error. Only Degraded should mark error.
	if req.UpgradeMode {
		return req.conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
	}

return !conditions.IsStatusConditionTrue(hcov1beta1.ConditionAvailable) || conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
}
```